canonicalwebteam/blog_extension/views.py

```diff
@@ -0,0 +1,165 @@
+import flask
+
+from webapp.blog_ext import api
+from webapp.blog_ext import logic
+
+
+def build_blueprint(blog_title, tags_id, tag_name):
+    blog = flask.Blueprint(
+        "blog", __name__, template_folder="/templates", static_folder="/static"
+    )
+
+    @blog.route("/")
+    def homepage():
+        page_param = flask.request.args.get("page", default=1, type=int)
+        filter = flask.request.args.get("filter", default=None, type=str)
+
+        if filter == "all":
+            filter = None
+
+        try:
+            categories_list = api.get_categories()
+        except Exception:
+            categories_list = []
+
+        categories = logic.whitelist_categories(categories_list)
+
+        filter_category = next(
+            (
+                item["id"]
+                for item in categories
+                if item["name"].lower() == filter
+            ),
+            None,
+        )
+
+        try:
+            articles, total_pages = api.get_articles(
+                tags=tags_id, page=page_param, category=filter_category
+            )
+        except Exception as e:
+            print(e)
```

Comment:
Use a logger here instead?

Reply:
So this doesn't take too long to be merged, I've agreed with Toto that we defer anything more than core fixes to future issues. 

So I've filed an issue for this point - https://github.com/canonical-webteam/blog-flask-extension/issues/4 - and I'll resolve this conversation. Hope that's okay.

Reply:
this is my mistake, removing it now